operator/scheduler/server.go

```diff
@@ -23,40 +23,54 @@ import (
 	"github.com/seldonio/seldon-core/operator/v2/apis/mlops/v1alpha1"
 )
 
-func (s *SchedulerClient) ServerNotify(ctx context.Context, server *v1alpha1.Server) error {
+func (s *SchedulerClient) ServerNotify(ctx context.Context, grpcClient scheduler.SchedulerClient, servers []v1alpha1.Server) error {
 	logger := s.logger.WithName("NotifyServer")
-	conn, err := s.getConnection(server.Namespace)
-	if err != nil {
-		return err
+	if len(servers) == 0 {
+		return nil
 	}
-	grpcClient := scheduler.NewSchedulerClient(conn)
-
-	var replicas int32
-	if !server.ObjectMeta.DeletionTimestamp.IsZero() {
-		replicas = 0
-	} else if server.Spec.Replicas != nil {
-		replicas = *server.Spec.Replicas
-	} else {
-		replicas = 1
+
+	if grpcClient == nil {
+		// we assume that all servers are in the same namespace
```

Comment:
All servers associated with a given scheduler (connection) will be in the same namespace, right? As in, this is always a safe assumption to make under the current design, as far as I know. Useful to have the comment though in case we change things in the future.